projects/plugins/boost/app/assets/src/js/pages/critical-css-advanced/critical-css-advanced.tsx

```diff
@@ -136,25 +154,55 @@ const HeadingMeta = ( { dismissedIssues, showDismissedIssues }: HeadingMetaProps
 	);
 };
 
-const Recommendation = ( { provider, setDismissed }: RecommendationProps ) => {
-	if ( provider.errors && provider.errors.length === 0 ) {
+const Recommendation = ( { recommendation, setDismissed }: RecommendationProps ) => {
+	if ( recommendation.errors && recommendation.errors.length === 0 ) {
 		return null;
 	}
 
-	const errorSet = getPrimaryGroupedError( provider.errors ? provider.errors : [] );
-	if ( ! errorSet ) {
+	const errorSets = groupErrorsByFrequency( recommendation.errors ? recommendation.errors : [] );
+	if ( errorSets.length === 0 ) {
 		return null;
 	}
 
-	const [ isDismissed, setIsDismissed ] = useState( provider.error_status === 'dismissed' );
+	return (
+		<>
+			{ errorSets.map( ( errorSet, _index ) => (
+				<SingleRecommendation
+					key={ `${ recommendation.key }-${ errorSet.type }` }
+					recommendation={ recommendation }
+					errorSet={ errorSet }
+					setDismissed={ setDismissed }
+				/>
+			) ) }
+		</>
+	);
```

Comment:
This could be simplified to the below:
```suggestion
	return errorSets.map( ( errorSet, _index ) => (
		<SingleRecommendation
			key={ `${ recommendation.key }-${ errorSet.type }` }
			recommendation={ recommendation }
			errorSet={ errorSet }
			setDismissed={ setDismissed }
		/>
	) );
```

Reply:
Oops. Nice catch 😅